commands/deploy_file.js

```diff
@@ -37,50 +36,36 @@ exports.builder = {
 }
 
 exports.handler = async function (argv) {
-    const content = fs.readFileSync(argv.file)
-    
-    transaction = null
-    txBuilder = archethic.newTransactionBuilder("hosting")
-        .setContent(content)
-    
+
+    setcontent(argv.file)
+
     const address = archethic.deriveAddress(argv.seed, 0)
-    
-    try {
-        index = await archethic.getTransactionIndex(address, argv.endpoint)
-    } catch (e) {
-        console.error(chalk.red(e.message))
-        return
-    }
 
-    transaction = txBuilder
-        .build(argv.seed, index)
-        .originSign(originPrivateKey)
+    index = await getindex(address, argv.endpoint)
+
+    transaction = buildtxn(argv.seed, index)
 
 
     try {
-        const { fee: fee, rates: rates } = await archethic.getTransactionFee(transaction, argv.endpoint)
-        
-        const ok = await yesno({
-            question: chalk.yellow('The transaction would cost ' +fee+ ' UCO ($ ' +rates.usd+ ' € ' +rates.eur+ '). Do you want to confirm ?')
-        });
-
-        if (ok)
-        {
-        archethic.waitConfirmations(transaction.address, argv.endpoint, function(nbConfirmations) {
-            if(nbConfirmations == 1)
-            {
-                console.log(chalk.green("Transaction Sent Successfully !"))
-                console.log(chalk.blue(argv.endpoint + "/api/last_transaction/" + (toHex(transaction.address)) + "/content?mime=" + mime.getType(argv.file)))
-            }
-            console.log(chalk.magenta("Transaction confirmed with " + nbConfirmations + " replications"))
-        })
-
-        send_file = await archethic.sendTransaction(transaction, argv.endpoint)
+
+        const ok = await feeconfirmation(transaction, argv.endpoint)
+
+        if (ok) {
+            archethic.waitConfirmations(transaction.address, argv.endpoint, function (nbConfirmations) {
+                if (nbConfirmations == 1) {
+                    console.log(chalk.green("Transaction Sent Successfully !"))
+                    console.log(chalk.blue(argv.endpoint + "/api/last_transaction/" + address + "/content?mime=" + mime.getType(argv.file)))
+                }
+                console.log(chalk.magenta("Transaction confirmed with " + nbConfirmations + " replications"))
+            })
```

Comment:
You can generate callback on confirmation like so... something like a function that returns another function what do you think
```js
archethic.waitConfirmations(transaction.address, argv.endpoint, confirmationCallback(endpoint, address))
            
// this returns the function itself that takes one param
function confirmationCallback(endpoint, address){
(nbConfirmations) => {
    if (nbConfirmations == 1) {
        console.log(chalk.green("Transaction Sent Successfully !"))
        console.log(chalk.blue(endpoint + "/api/last_transaction/" + address + "/content?mime=" + mime.getType(argv.file)))
    }
    console.log(chalk.magenta("Transaction confirmed with " + nbConfirmations + " replications"))
})
 }
}
```